src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_QuantumForceTransformer.java

```diff
@@ -0,0 +1,1160 @@
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .addElement(
+                            'Z',
+                            buildHatchAdder(GregtechMetaTileEntity_QuantumForceTransformer.class)
+                                    .hatchClass(GT_MetaTileEntity_Hatch_Input.class)
+                                    .adder(GregtechMetaTileEntity_QuantumForceTransformer::addNeptuniumHatch)
+                                    .casingIndex(TAE.getIndexFromPage(0, 10))
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .addElement(
+                            'X',
+                            buildHatchAdder(GregtechMetaTileEntity_QuantumForceTransformer.class)
+                                    .hatchClass(GT_MetaTileEntity_Hatch_Input.class)
+                                    .adder(GregtechMetaTileEntity_QuantumForceTransformer::addFermiumHatch)
+                                    .casingIndex(TAE.getIndexFromPage(0, 10))
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .build();
+        }
+        return this.STRUCTURE_DEFINITION;
+    }
+
+    public final boolean addCatalystHousing(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) {
+            return false;
+        } else {
+            IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+            if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Catalysts) {
+                return addToMachineList(aTileEntity, aBaseCasingIndex);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean addToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        final IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+        if (aMetaTileEntity == null) {
+            return false;
+        }
+        if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Catalysts) {
+            log("Found GT_MetaTileEntity_Hatch_Catalysts");
+            return addToMachineListInternal(mCatalystBuses, aMetaTileEntity, aBaseCasingIndex);
+        }
+        return super.addToMachineList(aTileEntity, aBaseCasingIndex);
+    }
+
+    @Override
+    public boolean checkMachine(IGregTechTileEntity aBaseMetaTileEntity, ItemStack aStack) {
+        this.mCasing = 0;
+        this.mMinimumMufflerTier = 0;
+        mCatalystBuses.clear();
+        if (checkPiece(this.mName, 7, 20, 4)
+                && checkHatch()
+                && (mTecTechEnergyHatches.size() >= 1 || mEnergyHatches.size() >= 1)) {
```

Comment:
A nitpick but if you already want to check if enough hatches are present, why not also check for Maintenance hatches.

Reply:
Will do it. I hate it when multis form without every min requirement.